src/tools/bubblechart/bubblechart-panzoom.js

```diff
@@ -249,11 +249,37 @@ export default Class.extend({
         var radiusMax = utils.areaToRadius(_this.sScale(_this.xyMaxMinMean.s[_this.timeFormatter(_this.time)].max));
         var frame = _this.currentZoomFrameXY;
 
+        var pan = this.zoomer.translate();
+        var zoom = this.zoomer.scale();
+        var ratioX = this.zoomer.ratioX;
+        var ratioY = this.zoomer.ratioY;
+
+        if(pan[0] > 0) pan[0] = 0;
```

Comment:
The lines 257-263 and 266-267 are copied from the `zoom.go()` function. Is there a preferred method of refactoring these lines? Note that the `xRange`, `yRange`, `xRangeBumped`, and `yRangeBumped` objects are needed later in the `zoom.go()` function.